pkg/sink/codec/open/open_protocol_encoder.go

```diff
@@ -37,14 +37,47 @@ type BatchEncoder struct {
 	config *common.Config
 }
 
+func (d *BatchEncoder) buildMessageOnlyHandleKeyColumns(e *model.RowChangedEvent) ([]byte, []byte, error) {
+	keyMsg, valueMsg, err := rowChangeToMsg(e, d.config.DeleteOnlyHandleKeyColumns, true)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	key, err := keyMsg.Encode()
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	value, err := valueMsg.encode(d.config.OnlyOutputUpdatedColumns)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+
+	length := len(key) + len(value) + common.MaxRecordOverhead + 16 + 8
+	if length > d.config.MaxMessageBytes {
+		log.Warn("Single message is too large for open-protocol",
+			zap.Int("maxMessageBytes", d.config.MaxMessageBytes),
+			zap.Int("length", length),
+			zap.Any("table", e.Table),
+			zap.Any("key", key))
+		return nil, nil, cerror.ErrMessageTooLarge.GenWithStackByArgs()
```

Comment:
Why are two nils being used here?

Reply:
since return error here, 2 nils means that key and value are nil. It's ok to return nils here.